src/modules/aws_opensearch/mappers/domain.ts

```diff
@@ -0,0 +1,398 @@
+          },
+          OffPeakWindowOptions: {
+            Enabled: true,
+            OffPeakWindow: { WindowStartTime: { Hours: 0, Minutes: 0 } }, // TODO: maybe input?
+          },
+          SoftwareUpdateOptions: { AutoSoftwareUpdateEnabled: true },
+        };
+        if (e.availabilityZoneCount > 1)
+          input.ClusterConfig!.ZoneAwarenessConfig = {
+            AvailabilityZoneCount: e.availabilityZoneCount,
+          };
+        if (!!e.subnets?.length && !!e.securityGroups?.length)
+          input.VPCOptions = {
+            SubnetIds: e.subnets.map(s => s.subnetId!),
+            SecurityGroupIds: e.securityGroups.map(sg => sg.groupId!),
+          };
+        if (e.ebsOptions) input.EBSOptions = e.ebsOptions;
+        if (e.customEndpoint && e.customEndpointCertificate)
+          input.DomainEndpointOptions = {
+            EnforceHTTPS: true,
+            TLSSecurityPolicy: 'Policy-Min-TLS-1-0-2019-07',
+            CustomEndpointEnabled: !!e.customEndpoint,
+            CustomEndpoint: e.customEndpoint,
+            CustomEndpointCertificateArn: e.customEndpointCertificate.arn,
+          };
+        if (
+          (e.fineGrainedAccessControlMasterUsername && e.fineGrainedAccessControlMasterPassword) ||
+          e.fineGrainedAccessControlUserArn
+        ) {
+          input.DomainEndpointOptions = input.DomainEndpointOptions ?? {};
+          input.DomainEndpointOptions.EnforceHTTPS = true;
+          input.DomainEndpointOptions.TLSSecurityPolicy = 'Policy-Min-TLS-1-0-2019-07';
+
+          input.AdvancedSecurityOptions = {
+            Enabled: true,
+            InternalUserDatabaseEnabled: !e.fineGrainedAccessControlUserArn,
+          };
+          if (e.fineGrainedAccessControlUserArn)
+            input.AdvancedSecurityOptions.MasterUserOptions = {
+              MasterUserARN: e.fineGrainedAccessControlUserArn,
+            };
+          else
+            input.AdvancedSecurityOptions.MasterUserOptions = {
+              MasterUserName: e.fineGrainedAccessControlMasterUsername,
+              MasterUserPassword: e.fineGrainedAccessControlMasterPassword,
+            };
+        }
+        // TODO: Cognito
+        await client.opensearchClient.createDomain(input);
+        // wait for it to become available
+        await this.waitForUpdate(client.opensearchClient, e.domainName);
+        const rawDomain = await client.opensearchClient.describeDomain({ DomainName: e.domainName });
+        if (!rawDomain || !rawDomain.DomainStatus) continue;
+        // now let's replace values with cloud ones to avoid infinite loop
+        const domain = await this.domainMapper(rawDomain.DomainStatus, e.region, ctx);
+        if (!domain) continue;
+        e.endpoint = domain.endpoint;
+        e.fineGrainedAccessControlMasterUsername =
+          e.fineGrainedAccessControlMasterPassword =
+          e.fineGrainedAccessControlUserArn =
+            undefined;
+        await this.module.domain.db.update(e, ctx);
```

Comment:
Maybe it is better to do something like the following to work with the latest object?
```suggestion
        const domain = await this.domainMapper(rawDomain.DomainStatus, e.region, ctx);
        if (!domain) continue;
        domain.id = e.id;
        domain.fineGrainedAccessControlMasterUsername =
          domain.fineGrainedAccessControlMasterPassword =
          domain.fineGrainedAccessControlUserArn =
            undefined;
        await this.module.domain.db.update(domain, ctx);
```